deepchecks/utils/metrics.py

```diff
@@ -118,29 +118,24 @@ def __init__(self, scorer: t.Union[str, t.Callable], name: str):
             raise errors.DeepchecksValueError(message)
 
     @classmethod
-    def filter_nulls(cls, dataset: 'tabular.Dataset'):
+    def filter_nulls(cls, dataset: 'tabular.Dataset') -> 'tabular.Dataset':
         """Return data of dataset without null labels."""
         valid_idx = dataset.data[dataset.label_name].notna()
-        return dataset.data[valid_idx]
+        return dataset.copy(dataset.data[valid_idx])
 
-    def _run_score(self, model, dataframe, dataset):
```

Comment:
I seem to remember there was a reason for that... places where the dataframe passes was not the dataset.feature_columns. Could you please check?